library/std/src/io/copy.rs

```diff
@@ -56,12 +56,47 @@ where
 
 /// The general read-write-loop implementation of
 /// `io::copy` that is used when specializations are not available or not applicable.
-pub(crate) fn generic_copy<R: ?Sized, W: ?Sized>(reader: &mut R, writer: &mut W) -> io::Result<u64>
+pub(crate) fn generic_copy<R: ?Sized, W: ?Sized>(reader: &mut R, writer: &mut W) -> Result<u64>
```

Comment:
It seems strange for there to be specialization in the function used "when specializations are not available or not applicable".

Reply:
That refers to another set of specializations. I'll try to reword it.